tests/core/test_plan_stages.py

```diff
@@ -1936,6 +1936,76 @@ def test_build_plan_stages_virtual_environment_mode_no_updates(
     assert len(virtual_stages) == 0
 
 
+def test_build_plan_stages_virtual_environment_mode_metadata_only_changes(
```

Comment:
Let's also add an integration test which involves a forward-only change to a view and check that the schema changes get reflected even though there are no evaluations.